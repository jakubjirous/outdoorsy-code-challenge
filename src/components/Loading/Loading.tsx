import React from "react";
import styled from "styled-components";
import { pxToRem } from "../../utils/pxToRem";
import { IMAGE_HEIGHT, IMAGE_WIDTH } from "../Image/Image";
import { Skeleton } from "../Skeleton/Skeleton";

const S = {
  Wrapper: styled.div`
    display: flex;
    flex-flow: column;
    gap: var(--spacing-6);
  `,
  Skeletons: styled.div`
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    gap: var(--spacing-5);
  `,
};

interface Props {
  numberOfSkeletons?: number;
}

function Loading({ numberOfSkeletons = 1 }: Props) {
  return (
    <S.Wrapper>
      {Array.from(Array(numberOfSkeletons).keys()).map((index) => (
        <S.Skeletons key={index}>
          <Skeleton width={pxToRem(IMAGE_WIDTH)} height={pxToRem(IMAGE_HEIGHT)} />
          <Skeleton width="50%" height="var(--fs-lg)" />
        </S.Skeletons>
      ))}
    </S.Wrapper>
  );
}

export default Loading;
