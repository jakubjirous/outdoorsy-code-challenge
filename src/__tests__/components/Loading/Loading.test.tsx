import { render } from "@testing-library/react";
import Loading from "../../../components/Loading/Loading";

const renderContainer = () => {
  return render(<Loading />);
};

describe("<Loading/>", () => {
  test("should render correctly", () => {
    const { baseElement } = renderContainer();

    expect(baseElement).toMatchSnapshot();
  });
});
