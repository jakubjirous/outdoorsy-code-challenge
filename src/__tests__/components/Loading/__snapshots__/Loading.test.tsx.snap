// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> should render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-jSMfEi beGjEx"
    >
      <div
        class="sc-gKXOVf gnvyZk"
      >
        <div
          class="sc-bczRLJ kkcSEP"
          height="8.4375rem"
          width="11.25rem"
        />
        <div
          class="sc-bczRLJ fNpSxG"
          height="var(--fs-lg)"
          width="50%"
        />
      </div>
    </div>
  </div>
</body>
`;
